### ISSUE TITLE
empty error message when sending due to "database disk image is malformed"

### ISSUE THREAD

**Issue by riyapenn** [signalapp/Signal-iOS#1677]
- [X ] I have searched open and closed issues for duplicates, but think there is an image referenced in another issue or pull request that I can not currently find
- Root causes for this specific alert: https://github.com/WhisperSystems/Signal-iOS/issues/1415 
----------------------------------------

### Bug description
Text not displayed in the alert. Tapping on alert does not yield any prompt for the user. 

### Steps to reproduce
- Phone languages is set to Hungarian
- Send message to contact

**Actual result:** See outline for an alert but not text of what it is
**Expected result:** See text in alert, which then can lead user to take steps to resolve (and ask contact to resend previous message)

### Screenshots

<img width="215" alt="screen shot 2017-01-31 at 2 23 36 pm" src="https://cloud.githubusercontent.com/assets/4898366/22487569/c850fbee-e7c2-11e6-9c3c-40305840e4b3.png">

### Device info

**Device**: iPhone SE
**iOS version**: 10.2.1
**Signal version:** 2.6.14

### Link to debug log

 https://gist.github.com/2525dd027595f6ce6153d4fa5aca4f6d

**Comment by michaelkirk** [comment]
This user is not experiencing an out-of-sync encryption session (#1415), rather, it seems their database has broken at a really low level. I suspect the reason the error box is empty is because we aren't handling it, as I don't recall ever having seen this before.

It would be interesting to know if the user could successfully message any contact. Based on the error, I'd assume they couldn't message anyone or record any new messages.

Key log message:

> 2017/01/31 14:02:50:730  YapDatabaseTransaction: Error executing 'insertForRowidStatement': 11 database disk image is malformed

@riyapenn Please update this issue with any further debug logs that include that logging line.

### Developer Notes

Upstream issue: https://github.com/yapstudios/YapDatabase/issues/81

Basically, the sqlite instance itself is corrupt. Here are some possible causes for us to investigate: http://www.sqlite.org/howtocorrupt.html

**Comment by riyapenn** [comment]
Seeing the same UIUX result for a different reason, so this may be a separate issue?

### Screenshots

<img width="292" alt="screen shot 2017-06-21 at 12 45 20 pm" src="https://user-images.githubusercontent.com/4898366/27403507-8acda6a6-567f-11e7-8fbc-6a62bc3bb064.png">



### Device info

**Device**: iPhone 6S
**iOS version**: 10.2.1
**Signal version:** 2.12.2

### Link to debug log


 https://gist.github.com/80ce462528683d5dd01d5b5f0aa40f65

**Comment by vstarlinger** [comment]
I am experiencing a very similar bug (however I do not see any error bubbles) with the "2017/07/28 18:28:10:844  YapDatabaseTransaction: Error executing 'insertForRowidStatement': 11 database disk image is malformed" error.

I no longer get new messages displayed in the app itself (old messages however do get displayed so no point in posting a screenshot because everything looks normal (only old)). Messages however do get displayed in the notification center. 

I can also send messages successfully (and also see and read them without any problems on the signal desktop app). However sending attachments doesn't work.

I'm using an iPhone 5S (language set to English).

Is there any hope in fixing this issue without re-installing the Signal App (and deleting all the old messages), Thanks

Link to debug log:
https://gist.github.com/anonymous/f857fecdf517940c1ff04c9be2050304

**Comment by michaelkirk** [comment]
@vstarlinger - did you do some kind of system restore? Or did this just start happening out of the blue?

When was the last time you were able to receive messages?

> I can also send messages successfully (and also see and read them without any problems on the signal desktop app). However sending attachments doesn't work.

It's interesting that you can do this - since this entails writing to disk to mutate your session.

Can you check out your disk usage for me?

Go to Settings.app > General > Storage > Manage Storage

What is your Used vs. Available space?
And how much is Signal using?

**Comment by vstarlinger** [comment]
@michaelkirk - no I didn't do any system restore. It happened just out of the blue. The last messages that get displayed are from July 21st 04:30 UTC+2

I do still receive the messages on the phone and they get displayed in the notification center (and the time in the Signal Inbox shows the correct time of the last message as well) -> see screenshots

![img_3994](https://user-images.githubusercontent.com/3217613/28746143-8ab4b450-7486-11e7-995f-e1454d5ffe7a.jpeg)
![img_3995](https://user-images.githubusercontent.com/3217613/28746144-94ab9e92-7486-11e7-869b-ab406112cf9d.jpeg)

Disk usage:
        Used: 8.98 GB
        Available: 3.14 GB

Signal: 566.9 MB (Documents & Data 535.4 MB)

Log with sent message: https://gist.github.com/anonymous/91032c8cbfce4bf221fbadf1e198408f

**Comment by darrenchaker** [comment]
Excellent resoluiton on this issue, and thank each of you for your efforts as I too was continuing to run into similar issues. Best, [Darren Chaker](www.blogdarrenchaker.com)

**Comment by gp2015** [comment]
This error started happening to me after updating to OS X 10.14.3. Macbook battery hit 0% while Signal was open and device was asleep.
